### ISSUE TITLE
Data Source Manager does not open anymore on Linux after adding any raster layer.

### ISSUE THREAD

**Issue by danschef** [qgis/QGIS#51520]
### What is the bug or the crash?

The Data Source Manager does not open again after adding any raster layer in QGIS 3.28.2 installed within MambaForge on Ubuntu 20.04. No matter if I press the button in the toolbar or if I try to open it via the "Layer" menu, the Data Source Manager does not show up again.

### Steps to reproduce the issue

Create a new conda environment within MambaForge and start QGIS:

```
mamba create -n qgis -c conda-forge "qgis=3.28.2"
mamba activate qgis
qgis  # to start QGIS from the command line
```

Press the "Open Data Source Manager" button on the toolbar and add a raster layer. Close the Data Source Manager window and press the "Open Data Source Manager" button again. Nothing happens.

### Versions

- QGIS 3.28.2
- Ubuntu 20.04

### Supported QGIS version

- [X] I'm running a supported QGIS version according to the roadmap.

### New profile

- [X] I tried with a new QGIS profile

**Comment by danschef** [comment]
Yes, I tried a new QGIS profile, same problem there. I need this to run within a conda-forge environment, that´s why I use MambaForge. The conda-forge version of QGIS is packaged by the [qgis-feedstock](https://github.com/conda-forge/qgis-feedstock) developers. Do you have any clue what might be wrong there and where they could have a look at? Probably, you know what happens in the QGIS backend when starting the Data Source Manager or where I could find some debugging information.

**Comment by gillins** [comment]
Can you reproduce this @SrNetoChan ? Seems fine for me on Ubuntu 22.04.

**Comment by danschef** [comment]
Also after upgrading from Ubuntu 20.04 to 22.04, the issue persists.

**Comment by danschef** [comment]
Somehow it seems to be related to the X11 forwarding of MobaXTerm. I am starting QGIS on a remote server where I am connected via MobaXTerm (22.2). If I use  XShell instead, everything works fine. However, it is still strange, because I had no issues with older QGIS releases (also in the same setup).

**Comment by gillins** [comment]
The Data Source Manager is a bit weird since it is a 'modeless' dialog - ie you can click back on the main window again while it is open. I've seen weird things with X clients and modeless dialogs before.... We've recently bumped the version of `qt` that `qgis` links against which may explain the differences you are seeing. 
Does the same thing happen with the "DB Manager" (also modeless)?

**Comment by danschef** [comment]
As long as I only open and close the DB Manager, everything works fine. But that is the same with the Data Source Manager - only after adding a layer, it does not show up anymore. I have to check if that is also the case for the DB Manager but currently, I have no DB at hand.

**Comment by danschef** [comment]
Ok, I played a bit around with the X11 settings of MobaXTerm and the issue is definitely the X-Server. I dicovered that somehow MobaXTerm was using an old Xorg version which is probably not compatible with the new `qt`.

![grafik](https://user-images.githubusercontent.com/16669114/215465458-8fdc0e96-60ec-4400-b83c-85df7331cdcc.png)

I switched the Xorg version from "MobaX_1.16.3" to "MobaX", which I assume is the latest one. Now everything works fine. Thanks all for looking at this.
